### PR TITLE
[DGL-LifeSci] Migration and Refactor

### PR DESCRIPTION
**Pull request by mufeili**
## Description

- Refactored the previous chemistry model zoo and made it a separate package DGL-LifeSci.
- Decompose models into two levels (GNNs/Readouts) for flexibility.
- Added test cases.
- Added deprecation messages.
- Migrated examples.

## Organization
```
apps/
    life_sci/
        dglls/
            data/
            model/
                gnn/
                readout/
                model_zoo/
                pretrain.py
           utils/
        examples/
        tests/
```

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

## Migration Guide

- Datasets previously under `dgl.data.chem.datasets` are now under `dglls.data`
- Utils for data loading/graph construction/featurization are now under `dglls.utils`
- Models previously under `dgl.model_zoo.chem` are now under `dglls.model`

## Potential To-dos

- [ ] Add CI
- [ ] Add Support for Conda/Pip installation
- [ ] Add website page for hosting the documentation

### CODE REVIEW

**Comment by BarclayII** [comment]
`dglls` looks better than `gls`; the latter is too opaque and doesn't sound like a DGL package (and PyPI unfortunately already has one `gls` package) (maybe `dgl_ls` or `dgl_lifesci` is even better).

**Comment by VoVAllen** [comment]
I don't think it's necessary to contain dgl in name since pytorch_geometric also used pyg as abbr. Usually people use <3 letters for abbr, or I think `dgl_ls` following python name convention looks better than `dglls`

**Comment by mufeili** [comment]
@BarclayII @VoVAllen 
For package naming, I think the goals are:
1. Emphasize that this is a DGL-based package.
2. Make the name easy to remember and interpret
3. Follow existing practices, e.g. torchvision, gluoncv, gluonnlp

For these reasons, I think dglls is fine. If this is still too opaque, we can also go with dgllife.

**Comment by mufeili** [comment]
@VoVAllen What kind of decorator are you suggesting for deprecation?

**Comment by VoVAllen** [comment]
https://stackoverflow.com/a/48632082/12552906
```python
import warnings

def deprecated(message):
  def deprecated_decorator(func):
      def deprecated_func(*args, **kwargs):
          warnings.warn("{} is a deprecated function. {}".format(func.__name__, message),
                        category=DeprecationWarning,
                        stacklevel=2)
          warnings.simplefilter('default', DeprecationWarning)
          return func(*args, **kwargs)
      return deprecated_func
  return deprecated_decorator


@deprecated("Use method yyy instead")
def some_method()"
 pass

```

File: apps/life_sci/README.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+mols = []
+for i in range(4):
+    SMILES = model(rdkit_mol=True)
+    mols.append(Chem.MolFromSmiles(SMILES))
+# Generating 4 molecules takes less than a second.
+
+SVG(Draw.MolsToGridImage(mols, molsPerRow=4, subImgSize=(180, 150), useSVG=True))
+```
+
+![](https://s3.us-east-2.amazonaws.com/dgl.ai/model_zoo/drug_discovery/dgmg_model_zoo_example2.png)
```
[the remark this answers is not in the document]
Comment (mufeili) [reply apps/life_sci/README.md]: Done.

File: apps/life_sci/dglls/data/alchemy.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+        if not os.path.exists(str(self.file_dir)):
+            archive = zipfile.ZipFile(self.zip_file_path)
+            archive.extractall(file_dir)
+            archive.close()
+
+        self._load(mol_to_graph, node_featurizer, edge_featurizer)
+
+    def _load(self, mol_to_graph, node_featurizer, edge_featurizer):
+        if self.load:
+            self.graphs, label_dict = load_graphs(osp.join(self.file_dir, "%s_graphs.bin" % self.mode))
```
[the remark this answers is not in the document]
Comment (mufeili) [reply apps/life_sci/dglls/data/alchemy.py]: Done.

**Comment by VoVAllen** [comment]
I prefer dgllife if considering the name convention of gluoncv/gluonnlp

File: apps/life_sci/dglls/model/model_zoo/gat_predictor.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        By default, ELU is applied for intermediate GAT layers and no activation is applied
+        for the last GAT layer.
+    classifier_hidden_feats : int
+        Size of hidden graph representations in the classifier. Default to 128.
+    classifier_dropout : float
+        The probability for dropout in the classifier. Default to 0.
+    n_tasks : int
+        Number of tasks, which is also the output size. Default to 1.
+    """
+    def __init__(self, in_feats, hidden_feats, num_heads, feat_drops=None, attn_drops=None,
```
[the remark this answers is not in the document]
Comment (mufeili) [reply apps/life_sci/dglls/model/model_zoo/gat_predictor.py]: Done.

File: apps/life_sci/dglls/model/model_zoo/gcn_predictor.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        ``len(dropout)`` equals the number of GCN layers. By default, no dropout is
+        performed for all layers.
+    classifier_hidden_feats : int
+        Size of hidden graph representations in the classifier. Default to 128.
+    classifier_dropout : float
+        The probability for dropout in the classifier. Default to 0.
+    n_tasks : int
+        Number of tasks, which is also the output size. Default to 1.
+    """
+    def __init__(self, in_feats, hidden_feats, activation=None, residual=None, batchnorm=None,
```
[the remark this answers is not in the document]
Comment (mufeili) [reply apps/life_sci/dglls/model/model_zoo/gcn_predictor.py]: Done.

**Comment by mufeili** [comment]
@VoVAllen I am now using `dgllife` and the decorator you suggested.
